ext/ffi/ffi.c

```diff
@@ -1221,6 +1221,25 @@ static zend_result zend_ffi_cdata_cast_object(zend_object *readobj, zval *writeo
 }
 /* }}} */
 
+static int zend_ffi_should_convert_cdata(void)
+{
+	zend_execute_data *execute_data = EG(current_execute_data);
+
+	if (execute_data
+	 && EX(opline)
+	 && EX(call)
+	 && (EX(opline)->opcode == ZEND_FETCH_OBJ_R
+	  || EX(opline)->opcode == ZEND_FETCH_OBJ_FUNC_ARG
+	  || EX(opline)->opcode == ZEND_FETCH_DIM_R
+	  || EX(opline)->opcode == ZEND_FETCH_DIM_FUNC_ARG)
+	 && EX(call)->func
+	 && EX(call)->func->type == ZEND_INTERNAL_FUNCTION
+	 && EX(call)->func->internal_function.module == &ffi_module_entry) {
```

Comment:
Just a note that this will unfortunately not work with frameless calls, since they do not create a stack frame.

Reply:
Right. may be you see a better (simpler) possible test.

Ideally, we might add some flag to ``opline->extended_value`` for FETCH_DIM/OBJ_R/FUNC_ARG at compile time.
This is possible for frame-less functions, but not possible  for indirect calls. (``$f = "FFI::type"; $f($x)``).